### PR TITLE
Atlas v4

### PR DESCRIPTION
**Pull request by will-moore**
This upgrades OMERO.fpbioimage to use the current version 4 of FPBioImage.
The files are no-longer dynamically downloaded in the build because there is no release tag for version 4 of FPBioImage, but we still want to test and release a stable version for OMERO.fpbioimage (see discussion below).

Now, we host the PFBioimage binary files within the omero-fpbioimage repo and we need to update them when a new release is available.

We also use the "atlas" method of loading planes, which improves the speed of loading.
E.g. total time to sequentially load 60 planes of 500 * 500 goes from ~ 20 seconds to around 6 seconds to load 8 atlases simultaneously.

Total time from loading page to viewer ready goes from ~35 seconds with old viewer to ~16 seconds now. However, the feedback is not so good while atlases are loading: just says "Downloaded 0.0%" for a few secs then it's done.

cc @mfantham

### CODE REVIEW

File: omero_fpbioimage/views.py [hunk 1]
```diff
@@ -67,35 +67,70 @@ def fpbioimage(request, image_id, conn=None, **kwargs):
     return render(request, 'fpbioimage/viewer.html', context)
 
 
+def ceil2(x):
+    print "ceil", x, type(x)
```
Comment (jburel) [inline omero_fpbioimage/views.py]: no real need to keep the print statement

**Comment by jburel** [comment]
Loading the data is certainly much faster now
As you indicated, not much feedback, go from 0% to ready

**Comment by will-moore** [comment]
As discussed, there may be some advantages to hosting the FPBioimage binaries in OMERO.web:
 - If you are close to your OMERO.web server, but a long way from Cambridge (or even working offline with a local server, which is still possible with all other web apps) then it's better to have the files coming from OMERO.web
 - We can test that FPBioimage works with the current OMERO.FPBioimage code (and OMERO) to be sure that the version we release is working. If the version that users receive is dynamic, we can't be so sure that everything is going to work.

**Comment by will-moore** [comment]
To test last commits, simply check that the viewer still works with no errors (e.g. check dev tools console).

**Comment by jburel** [comment]
We need to indicate the version of the binary taken 
since there is no tag. Let's use the date YYYY-MM-DD and put it the changelog

**Comment by manics** [comment]
Tested with a fairly small image and it worked fine, though as pointed out in the description having the loading screen stuck at `Downloaded 0.0 %` isn't ideal.

I then tried importing a [lightsheet image from the IDR (1920 x 1920 x 292, 8.1GB)](https://idr.openmicroscopy.org/webclient/?show=image-3414052) into eel and opening it with FPBioimage in web-dev-merge.
- The first time a dialog popped up with this error: `An exception has occured, but exception handling has been disabled in this build. If you are the developer of this content, enable exceptions in your project WebGL player settings to be able to catch the exception or see the stack trace.`
- I retried an got this error in a popup dialog: `Out of memory. If you are the developer of this content, try allocating more memory to your WebGL build in the WebGL player settings.`
- CPU usage on web-dev-merge was 5 gunicorn processes at 100%:
```
 8251 hudson    20   0 1961156 1.174g  10944 R 100.3 15.4   1:04.31 gunicorn
 8269 hudson    20   0 1735036 1.144g  10588 R 100.3 15.0   1:39.59 gunicorn
 7896 hudson    20   0 1746236 1.155g  10604 R 100.0 15.1   3:39.88 gunicorn
 8182 hudson    20   0 1743432 1.152g  10524 R 100.0 15.1   2:18.19 gunicorn
 8250 hudson    20   0 1743932 1.151g  10604 R 100.0 15.1   1:03.63 gunicorn
```

**Comment by manics** [comment]
I've gone back through some Trello cards- we previously tried FPBioimage on lightsheet data in the IDR, similar to the image above that I used for testing. The conclusions were that loading planes took too long (10+ minutes), but FPBioimage worked. I've just tested the current release (0.2.0) and this is still the situation.

**Comment by will-moore** [comment]
@manics I'm now scaling down to max plane size of 512 * 512 (up to 512 slices), so bigger images should not exceed memory limits now (same behaviour as before).

**Comment by will-moore** [comment]
@manics Sorry - this isn't in today's build because of travis failure.

File: omero_fpbioimage/views.py [hunk 2]
```diff
@@ -59,43 +60,104 @@ def fpbioimage(request, image_id, conn=None, **kwargs):
...
+                         padded_slice_height + y_offset)
+        y_start_pixel = int(atlas_height - y_start_pixel -
+                            padded_slice_height + 2*y_offset)
+
+        # Render plane and paste onto atlas
+        jpeg_data = image.renderJpeg(the_z, 0, compression=0.9)
+        plane = Image.open(StringIO(jpeg_data))
+        if plane.size[0] > slice_width:
+            plane = plane.resize((slice_width, slice_height), Image.BICUBIC)
+            print plane.size
```
Comment (manics) [inline omero_fpbioimage/views.py]: You've got a `print` statement here

**Review by manics — COMMENTED** [review]
One minor issue (`print` statement).

I've tested this locally and it's looking at lot better, with the 512 size limit the lightsheet image I was using for testing successfully loads in Firefox.

It might be nice to have the limit configurable, but since it depends on fpbioimage that's probably best left for a future discussion.

**Comment by will-moore** [comment]
@jburel The "version 4" page https://github.com/fpBioImage/fpBioImage.github.io/tree/master/4 tells us that the last commit was https://github.com/fpBioImage/fpBioImage.github.io/commit/f0b4e312b85771f47bbe21e2bf6cb25b82b0f600, on 10th August so we can use that as the 'tag' or version date?
I'll add to changelog...

**Comment by jburel** [comment]
Merging, I will sort out date and version in another PR

**Pull request merged**